pkg/storage/mongo/getter.go

```diff
@@ -91,3 +63,30 @@ func (s *ModuleStore) Zip(ctx context.Context, module, vsn string) (io.ReadClose
 
 	return dStream, nil
 }
+
+// Query connects and queries storage module
+func Query(ctx context.Context, s *ModuleStore, module, vsn string, op errors.Op) (*storage.Module, error) {
```

Comment:
Can you have it as `func (s* moduleStore) query(ctx, module, .....)`  this way query will be private to the package?

Reply:
Definitely, that makes a lot more sense.